### PR TITLE
NIFI-12095 Increase default Xmx to 1g

### PR DESCRIPTION
**Pull request by nandorsoma**
# Summary

[NIFI-12095](https://issues.apache.org/jira/browse/NIFI-12095)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
     <properties>
         <!--Wrapper Properties -->
         <nifi.jvm.heap.init>512m</nifi.jvm.heap.init>
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/pom.xml:28]: The minimum heap size should be changed to match the maximum heap size.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for proposing this change @nandorsoma. Increasing the default heap size seems reasonable. As noted in the details, the minimum heap size should be changed to match the maximum.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/pom.xml (hunk 1 above)
Comment (nandorsoma) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/pom.xml:28]: Thanks for the review @exceptionfactory. Just out of curiosity, what is the reason for that? I thought a rule of thumb is to initially have the minimum as the 50% of the maximum heap size.

Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/pom.xml:28]: Setting the minimum heap size to the same as the maximum makes the overall process memory usage more predictable. There are other factors that contribute to total process memory usage, but heap is usually the primary contributor. Starting out with the same amount of heap as the maximum avoids unexpected surprises when the heap grows to handle cached objects.

**Review by exceptionfactory — APPROVED** [review]

**Comment by timeabarna** [comment]
LGTM, merging to main and support/nifi-1.x

**Comment by joewitt** [comment]
+1 merging to main and support/1.x

**Pull request closed without merging**